### PR TITLE
New Clique Minor Release Series

### PR DESCRIPTION
**Pull request by holgerd77**
Phew, this is somewhat more work than I expected, solely the `Common` release preparation took 1 1/2 hours. Anyhow.

Planning to prepare for the following releases here, all with a somewhat strong focus on the Clique PoA integration, there are other features to be release as well though

- [x] `Common` `v2.1.0`, fixes small typing bug from #1070 `dnsNetwork` -> `dnsNetworks`
- [x] `Block` `v3.1.0`, removed block.clique* function subset calling into header (inconsistent, not used, non-refactoring friendly), added two clique require checks
- [x] `Blockchain` `v5.1.0`
- [x] `VM` `v5.1.0`,  aligned `hardforkByBlockNumber` VM parameter with naming from Block (`selectHardforkByBlockNumber` -> `hardforkByBlockNumber`)

Review can be done per all libraries from above checked already or alternatively all at once.

### CODE REVIEW

**Comment by holgerd77** [comment]
Ok, added the `Block` release notes together with some small changes (see description above or commits).

**Comment by holgerd77** [comment]
Currently planning to do these releases on Monday or Tuesday.

File: packages/block/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,38 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
+This release introduces Clique/PoA support for the `Block` library, see the main PR [#1032](https://github.com/ethereumjs/ethereumjs-monorepo/pull/1032) as well as the follow-up PRs [#1074](https://github.com/ethereumjs/ethereumjs-monorepo/pull/1074) and PR [#1088](https://github.com/ethereumjs/ethereumjs-monorepo/pull/1088). The `BlockHeader.validate()` function now properly validates the various Clique/PoA-specific properties (`extraData` checks and others, see API documentation) and `BlockHeader.validataConsensus()` can be used to properly validate that a Clique/PoA block has the correct signature.
+
+For sealing a block on instantiation there is a new `cliqueSigner` constructor option:
+
+```typescript
+const cliqueSigner = Buffer.from('PRIVATE_KEY_HEX_STRING', 'hex')
+const block = Block.fromHeaderData(headerData, { cliqueSigner })
+```
+
+Additionally there are the following new utility methods for Clique/PoA relatec functionality in the `BlockHeader` class:
```
Comment (ryanio) [inline packages/block/CHANGELOG.md:22]:
```suggestion
Additionally there are the following new utility methods for Clique/PoA related functionality in the `BlockHeader` class:
```

File: packages/block/CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,38 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 (modification: no type change headlines) and this project adheres to
 [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## 3.1.0 - 2021-02-22
+
+### Clique/PoA Support
+
+This release introduces Clique/PoA support for the `Block` library, see the main PR [#1032](https://github.com/ethereumjs/ethereumjs-monorepo/pull/1032) as well as the follow-up PRs [#1074](https://github.com/ethereumjs/ethereumjs-monorepo/pull/1074) and PR [#1088](https://github.com/ethereumjs/ethereumjs-monorepo/pull/1088). The `BlockHeader.validate()` function now properly validates the various Clique/PoA-specific properties (`extraData` checks and others, see API documentation) and `BlockHeader.validataConsensus()` can be used to properly validate that a Clique/PoA block has the correct signature.
```
Comment (ryanio) [inline packages/block/CHANGELOG.md:13]:
```suggestion
This release introduces Clique/PoA support for the `Block` library, see the main PR [#1032](https://github.com/ethereumjs/ethereumjs-monorepo/pull/1032) as well as the follow-up PRs [#1074](https://github.com/ethereumjs/ethereumjs-monorepo/pull/1074) and PR [#1088](https://github.com/ethereumjs/ethereumjs-monorepo/pull/1088). The `BlockHeader.validate()` function now properly validates the various Clique/PoA-specific properties (`extraData` checks and others, see API documentation) and `BlockHeader.validateConsensus()` can be used to properly validate that a Clique/PoA block has the correct signature.
```

File: packages/block/README.md [hunk 3]
```diff
@@ -57,6 +57,38 @@ try {
...
+
+The block library supports the creation as well as format and consensus validation of PoW `ethash` and PoA `clique` blocks.
+
+## Ethash/PoW
+
+TODO
+
+## Clique/PoA (since v3.1.0)
+
+For clique PoA `BlockHeader.validate()` function validates the various Clique/PoA-specific properties (`extraData` checks and others, see API documentation) and `BlockHeader.validataConsensus()` can be used to properly validate that a Clique/PoA block has the correct signature.
```
Comment (ryanio) [inline packages/block/README.md:70]:
```suggestion
For clique PoA `BlockHeader.validate()` function validates the various Clique/PoA-specific properties (`extraData` checks and others, see API documentation) and `BlockHeader.validateConsensus()` can be used to properly validate that a Clique/PoA block has the correct signature.
```

File: packages/block/README.md [hunk 4]
```diff
@@ -57,6 +57,38 @@ try {
 
 [Documentation](./docs/README.md)
 
+# CONSENSUS TYPES
+
+The block library supports the creation as well as format and consensus validation of PoW `ethash` and PoA `clique` blocks.
+
+## Ethash/PoW
+
+TODO
```
Comment (ryanio) [inline packages/block/README.md:66]: a note/reminder we should include reference to the BlockOption `calcDifficultyFromHeader`

File: packages/block/README.md [hunk 5]
```diff
@@ -57,6 +57,38 @@ try {
...
+For clique PoA `BlockHeader.validate()` function validates the various Clique/PoA-specific properties (`extraData` checks and others, see API documentation) and `BlockHeader.validataConsensus()` can be used to properly validate that a Clique/PoA block has the correct signature.
+
+For sealing a block on instantiation you can use the `cliqueSigner` constructor option:
+
+```typescript
+const cliqueSigner = Buffer.from('PRIVATE_KEY_HEX_STRING', 'hex')
+const block = Block.fromHeaderData(headerData, { cliqueSigner })
+```
+
+Additionally there are the following utility methods for Clique/PoA relatec functionality in the `BlockHeader` class:
```
Comment (ryanio) [inline packages/block/README.md:79]:
```suggestion
Additionally there are the following utility methods for Clique/PoA related functionality in the `BlockHeader` class:
```

File: packages/block/README.md [hunk 6]
```diff
@@ -57,6 +57,38 @@ try {
...
+- `BlockHeader.validateCliqueDifficulty(blockchain: Blockchain): boolean`
+- `BlockHeader.cliqueSigHash()`
+- `BlockHeader.cliqueIsEpochTransition(): boolean`
+- `BlockHeader.cliqueExtraVanity(): Buffer`
+- `BlockHeader.cliqueExtraSeal(): Buffer`
+- `BlockHeader.cliqueEpochTransitionSigners(): Address[]`
+- `BlockHeader.cliqueVerifySignature(signerList: Address[]): boolean`
+- `BlockHeader.cliqueSigner(): Address`
+
+See the API docs for a detailed documentation. Note that these methods will throw if called in a non-Clique/PoA context.
```
Comment (ryanio) [inline packages/block/README.md:90]:
```suggestion
See the API docs for detailed documentation. Note that these methods will throw if called in a non-Clique/PoA context.
```

File: packages/block/README.md (hunk 4 above)
Comment (holgerd77) [reply packages/block/README.md:66]: Just had another look: `calcDifficultyFromHeader` has already been added to `v3.0.0`. 😄

**Comment by holgerd77** [comment]
Ok, all complete, this is now ready for review. 😄

File: packages/block/README.md [hunk 7]
```diff
@@ -57,6 +57,60 @@ try {
...
+```typescript
+import { Block } from '@ethereumjs/block'
+import Common from '@ethereumjs/common'
+const common = new Common({ chain: 'mainnet' })
+console.log(common.consensusType()) // 'pow'
+console.log(common.consensusAlgorithm()) // 'ethash'
+const block = Block.fromBlockData({}, { common })
+```
+
+TODO
```
Comment (ryanio) [inline packages/block/README.md:77]:
```suggestion
To validate that the difficulty of the block matches the canonical difficulty use `block.validate(blockchain)`.

To calculate the difficulty when creating the block pass in the block option `calcDifficultyFromHeader` with the preceding (parent) `BlockHeader`.
```

File: packages/vm/README.md [hunk 8]
```diff
@@ -75,6 +75,37 @@ To build the VM for standalone use in the browser, see: [Running the VM in a bro
 
 # SETUP
 
+## Chain Support
+
+Starting with `v5.1.0` the VM supports running both `Ethash/PoW` and `Clique/PoA` blocks and transactions. Clique support has been added along the work on PR [#1032](https://github.com/ethereumjs/ethereumjs-monorepo/pull/1032) and follow-up PRs and (block) validation checks and the switch of the execution context now happens correctly.
+
+### Ethash/PoW Chains
+
+TODO
```
Comment (ryanio) [inline packages/vm/README.md:84]:
```suggestion
`@ethereumjs/blockchain` validates the PoW algorithm with `@ethereumjs/ethash` and validates blocks' difficulty to match their canonical difficulty.
```

**Comment by holgerd77** [comment]
Have updated the `Common` changelog with the changes from #1125.

**Review by ryanio — APPROVED** [review]
great, lgtm!

**Comment by holgerd77** [comment]
Just published the following releases on npm:

- `@ethereumjs/common` [v2.1.0](https://github.com/ethereumjs/ethereumjs-monorepo/releases/tag/%40ethereumjs%2Fcommon%402.1.0)
- `@ethereumjs/block` [v3.1.0](https://github.com/ethereumjs/ethereumjs-monorepo/releases/tag/%40ethereumjs%2Fblock%403.1.0)
- `@ethereumjs/blockchain` [v5.1.0](https://github.com/ethereumjs/ethereumjs-monorepo/releases/tag/%40ethereumjs%2Fblockchain%405.1.0)
- `@ethereumjs/vm`[v5.1.0](https://github.com/ethereumjs/ethereumjs-monorepo/releases/tag/%40ethereumjs%2Fvm%405.1.0)

**Pull request merged**